src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs

```diff
@@ -92,6 +87,13 @@ public ClientConnectionContext(OpenConnectionMessage serviceMessage, Action<Http
             Application = pair.Transport;
 
             HttpContext = BuildHttpContext(serviceMessage);
+
+            if (serviceMessage.Headers.TryGetValue(Constants.AsrsMigrateIn, out var val))
+            {
+                IsMigrated = true;
+                HttpContext.Items.Add(Constants.AsrsMigrateIn, val);
```

Comment:
@davidfowl Didn't you want this on ConnectionContext.Items not HttpContext.Items?

```suggestion
                Items.Add(Constants.AsrsMigrateIn, val);
```

Reply:
Yea but now I want a feature instead.

Reply:
Feature looks good to me, maybe we make it `internal` first.

Reply:
Why would we make the feature internal? Are you talking about the feature interface itself?

Reply:
Right, it needs to be public.

Reply:
`internal` when it is not yet feature-ready, and switch to `public` when the feature is ready

Reply:
Oh sure, we shouldn't ship an SDK with this feature being internal, that's all I'm saying.